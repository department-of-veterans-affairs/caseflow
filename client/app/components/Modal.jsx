/* eslint-disable react/prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import ScrollLock from 'react-scrolllock';
import { closeSymbolHtml } from './RenderFunctions';
import Button from './Button';
import _ from 'lodash';
import { css } from 'glamor';

const modalTextStyling = css({ width: '100%',
  fontFamily: 'Source Sans Pro' });

export default class Modal extends React.Component {
  constructor(props) {
    super(props);
    this.buttonIdPrefix = `${this.props.title.replace(/\s/g, '-')}-button-id-`;
  }
  handleTab = (event) => {
    let buttonsList = document.querySelectorAll('.cf-modal-controls button');
    let firstButton = document.getElementById(`${this.buttonIdPrefix}close`);
    // a more general selector for the last button in the modall
    let endButton = buttonsList[buttonsList.length - 1];

    if (event.shiftKey) {
      if (firstButton === document.activeElement) {
        event.preventDefault();
        endButton.focus();
      }
    } else if (endButton === document.activeElement) {
      event.preventDefault();
      firstButton.focus();
    }
  };

  keyHandler = (event) => {
    if (event.key === 'Escape') {
      this.props.closeHandler();
    }

    if (event.key === 'Tab') {
      this.handleTab(event);
    }
  };

  modalCloseFocus = (modalClose) => (this.modalClose = modalClose);

  componentWillUnmount() {
    window.removeEventListener('keydown', this.keyHandler);
    // return focus to original target
    // document.querySelector('.cf-btn-link').focus();
  }

  componentDidMount() {
    window.addEventListener('keydown', this.keyHandler);
    this.modalClose.focus();
  }

  generateButtons() {
    return this.props.buttons.map((object, i) => {
      // If we have more than two buttons, push the
      // first left, and the rest right.
      // If we have just one button, push it right.
      let classNames = ['cf-push-right'];

      if (i === 0 && this.props.buttons.length > 1) {
        classNames = ['cf-push-left'];
      }

      if (typeof object.classNames !== 'undefined') {
        classNames = [...object.classNames, ...classNames];
      }

      return (
        <Button
          name={object.name}
          onClick={object.onClick}
          classNames={classNames}
          loading={object.loading}
          disabled={object.disabled}
          key={i}
          id={this.buttonIdPrefix + i}
        />
      );
    });
  }

  render() {
    const {
      children,
      className,
      closeHandler,
      id,
      noDivider,
      confirmButton,
      cancelButton,
      title,
      customStyles
    } = this.props;

    let modalButtons;

    if (!confirmButton && !cancelButton) {
      modalButtons = this.generateButtons();
    } else {
      modalButtons = (
        <div>
          <span className="cf-push-right">{confirmButton}</span>
          {cancelButton && <span className="cf-push-left">{cancelButton}</span>}
        </div>
      );
    }

    return (
      <section
        className={`cf-modal active ${className}`}
        id="modal_id"
        role="alertdialog"
        aria-labelledby="modal_id-title"
        aria-describedby="modal_id-desc"
        aria-modal="true"
      >
        <ScrollLock />
        <div className="cf-modal-body" id={id || ''} {...customStyles}>
          <button
            type="button"
            id={`${this.buttonIdPrefix}close`}
            className="cf-modal-close"
            onClick={closeHandler}
            ref={this.modalCloseFocus}
          >
            {closeSymbolHtml()}
          </button>
          <h1 id="modal_id-title">{title}</h1>
          <div {...modalTextStyling}>{children}</div>
          {noDivider ? '' : <div className="cf-modal-divider" />}
          <div className="cf-modal-controls">{modalButtons}</div>
        </div>
      </section>
    );
  }
}

Modal.defaultProps = {
  buttons: []
};

Modal.propTypes = {
  buttons: (props, propName) => {
    const buttons = props[propName];

    if (!_.isArray(buttons)) {
      return new Error(`'buttons' must be an array, but was: '${buttons}'`);
    }

    if (buttons.length && (props.cancelButton || props.confirmButton)) {
      return new Error('You cannot set both `buttons` and one of `confirmButton` or `cancelButton`');
    }
  },
  className: PropTypes.string,
  confirmButton: PropTypes.element,
  cancelButton: PropTypes.element,
  id: PropTypes.string,
  label: PropTypes.string,
  noDivider: PropTypes.bool,
  specialContent: PropTypes.func,
  title: PropTypes.string.isRequired
};

Modal.defaultProps = {
  className: ''
};
