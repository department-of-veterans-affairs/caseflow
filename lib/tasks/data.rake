# Requires the following environment variables to be set:
# - VACOLS_PASSWORD
# - VACOLS_USERNAME

namespace :data do
  staging_only_error = "This command only works in the staging environment. Please run with RAILS_ENV='staging'."

  desc "Prepare test data needed to run a smoke test"
  task prepare: [:environment] do
    fail staging_only_error unless Rails.env.staging?

    test_appeal_vacols_id = "2765748"

    test_appeal = Appeal.find_or_create_by_vacols_id(test_appeal_vacols_id)
    AppealRepository.uncertify(test_appeal)
  end
end
