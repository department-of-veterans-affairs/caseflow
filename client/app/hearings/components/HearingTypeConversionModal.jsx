import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { get } from 'lodash';
import { sprintf } from 'sprintf-js';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import React, { useState } from 'react';

import Modal from '../../components/Modal';
import { appealWithDetailSelector, taskById } from '../../queue/selectors';
import { clearAppealDetails } from '../../queue/QueueActions';
import { showErrorMessage, showSuccessMessage } from '../../queue/uiReducer/uiActions';
import ApiUtil from '../../util/ApiUtil';
import Button from '../../components/Button';
import { formatChangeRequestType } from '../utils';

// Constants
import COPY from '../../../COPY';
import TASK_STATUSES from '../../../constants/TASK_STATUSES';

export const HearingTypeConversionModal = ({
  appeal,
  history,
  task,
  hearingType,
  ...props
}) => {
  // Create and manage the loading state
  const [loading, setLoading] = useState(false);

  const getSuccessMsg = () => {
    // Format the message title adding the veteran info and hearing request type
    const title = sprintf(
      COPY.CONVERT_HEARING_TYPE_SUCCESS,
      appeal?.appellantIsNotVeteran ? appeal?.appellantFullName : appeal?.veteranFullName,
      hearingType
    );

    // Determine whether to use the default ro Label
    const defaultLabel = appeal?.closestRegionalOfficeLabel === 'Central Office' ?
      COPY.CONVERT_HEARING_TYPE_DEFAULT_REGIONAL_OFFICE_TEXT :
      appeal?.closestRegionalOfficeLabel;

    // Determine whether to override the default label if converting to Central Office
    const roLabel = hearingType === 'Central' ? 'Central Office' : defaultLabel;

    // Format the message details adding the appropriate RO label
    const detail = sprintf(COPY.CONVERT_HEARING_TYPE_SUCCESS_DETAIL, roLabel);

    return { title, detail };
  };

  const submit = async () => {
    try {
      // Determine the changed request type
      const changedRequestType = formatChangeRequestType(hearingType);

      // Determine the closest regional office
      const closestRegionalOffice = hearingType === 'Video' ? null : 'C';

      const data = {
        task: {
          status: TASK_STATUSES.completed,
          business_payloads: {
            values: {
              changed_hearing_request_type: changedRequestType,
              closest_regional_office: closestRegionalOffice
            }
          }
        }
      };

      setLoading(true);

      await ApiUtil.patch(`/tasks/${task.taskId}`, { data });

      // Add the google analytics event
      window.analyticsEvent('Hearings', 'Convert hearing request type', hearingType);

      props.clearAppealDetails(task.externalAppealId);
      props.showSuccessMessage(getSuccessMsg());
    } catch (err) {
      const error = get(
        err,
        'response.body.errors[0]',
        {
          title: COPY.DEFAULT_UPDATE_ERROR_MESSAGE_TITLE,
          detail: COPY.DEFAULT_UPDATE_ERROR_MESSAGE_DETAIL
        }
      );

      props.showErrorMessage(error);
    } finally {
      setLoading(false);

      history.push(`/queue/appeals/${appeal.externalId}`);
    }
  };

  const cancel = () => history.goBack();

  const convertTitle = sprintf(COPY.CONVERT_HEARING_TYPE_TITLE, hearingType);
  const convertSubtitle = sprintf(
    COPY.CONVERT_HEARING_TYPE_SUBTITLE,
    hearingType === 'Video' ?
      COPY.CONVERT_HEARING_TYPE_DEFAULT_REGIONAL_OFFICE_TEXT :
      '<strong>Central Office</strong>'
  );

  return (
    <Modal
      title={convertTitle}
      closeHandler={cancel}
      confirmButton={<Button disabled={loading} onClick={submit}>Convert Hearing to {hearingType}</Button>}
      cancelButton={<Button linkStyling disabled={loading} onClick={cancel}>Cancel</Button>}
    >
      <p dangerouslySetInnerHTML={{ __html: convertSubtitle }} />
    </Modal>
  );
};

HearingTypeConversionModal.propTypes = {
  appeal: PropTypes.object,
  appealId: PropTypes.string,
  clearAppealDetails: PropTypes.func,
  showErrorMessage: PropTypes.func,
  showSuccessMessage: PropTypes.func,
  task: PropTypes.object,
  taskId: PropTypes.string,
  hearingType: PropTypes.oneOf(['Video', 'Central']),
  // Router inherited props
  history: PropTypes.object
};

const mapStateToProps = (state, ownProps) => ({
  appeal: appealWithDetailSelector(state, ownProps),
  task: taskById(state, { taskId: ownProps.taskId })
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      clearAppealDetails,
      showErrorMessage,
      showSuccessMessage
    },
    dispatch
  );

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(HearingTypeConversionModal)
);
