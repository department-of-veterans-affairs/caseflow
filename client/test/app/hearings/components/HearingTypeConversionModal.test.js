import React from 'react';

import { mount } from 'enzyme';

import { appealData } from '../../../data/appeals';
import { queueWrapper } from '../../../data/stores/queueStore';
import HearingTypeConversionModal from '../../../../app/hearings/components/HearingTypeConversionModal';
import Modal from '../../../../app/components/Modal';
import COPY from 'COPY';

describe('HearingTypeConversion', () => {
  test('Displays convert to Video text when converting from Video', () => {
    const hearingTypeConversion = mount(
      <HearingTypeConversionModal
        appeal={appealData}
        hearingType="Video"
      />,
      {
        wrappingComponent: queueWrapper,
      }
    );

    expect(hearingTypeConversion.exists(Modal)).toBeTruthy();
    expect(hearingTypeConversion.find(Modal).text()).not.toContain('Central Office');
    expect(hearingTypeConversion.find(Modal).text()).toContain(COPY.CONVERT_HEARING_TYPE_DEFAULT_REGIONAL_OFFICE_TEXT);
    expect(hearingTypeConversion).toMatchSnapshot();
  });

  test('Displays convert to Central text when converting from Central', () => {
    const hearingTypeConversion = mount(
      <HearingTypeConversionModal
        appeal={appealData}
        hearingType="Central"
      />,
      {
        wrappingComponent: queueWrapper,
      }
    );

    expect(hearingTypeConversion.exists(Modal)).toBeTruthy();
    expect(hearingTypeConversion.find(Modal).text()).toContain('Central Office');
    expect(hearingTypeConversion.find(Modal).text()).
      not.toContain(COPY.CONVERT_HEARING_TYPE_DEFAULT_REGIONAL_OFFICE_TEXT);
    expect(hearingTypeConversion).toMatchSnapshot();
  });
});
