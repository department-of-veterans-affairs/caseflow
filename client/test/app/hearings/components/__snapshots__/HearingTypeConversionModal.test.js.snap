// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HearingTypeConversion Displays convert to Central text when converting from Central 1`] = `
<withRouter(Connect(HearingTypeConversionModal))
  appeal={
    Object {
      "appellantAddress": Object {
        "address_line_1": "9999 MISSION ST",
        "address_line_2": "UBER",
        "address_line_3": "APT 2",
        "city": "SAN FRANCISCO",
        "country": "USA",
        "state": "CA",
        "zip": "94103",
      },
      "appellantFullName": "Abellona Valtas",
      "appellantIsNotVeteran": false,
      "appellantRelationship": "Spouse",
      "appellantTz": "America/New_York",
      "assignedAttorney": null,
      "assignedJudge": null,
      "assignedToLocation": "Hearing Admin",
      "attorneyCaseRewriteDetails": Object {
        "note_from_attorney": null,
        "untimely_evidence": null,
      },
      "availableHearingLocations": Array [],
      "canEditDocumentId": false,
      "canEditRequestIssues": false,
      "caseReviewId": null,
      "caseType": "Original",
      "caseflowVeteranId": 541,
      "certificationDate": null,
      "closestRegionalOffice": null,
      "closestRegionalOfficeLabel": null,
      "completedHearingOnPreviousAppeal": false,
      "decisionDate": null,
      "decisionIssues": Array [
        Object {
          "description": "This is a description of the decision",
          "id": 1,
        },
        Object {
          "description": "This is a description of another decision",
          "id": 2,
        },
      ],
      "docketName": "hearing",
      "docketNumber": "200805-541",
      "documentID": null,
      "isAdvancedOnDocket": false,
      "isLegacyAppeal": false,
      "isPaperCase": false,
      "issueCount": 0,
      "issues": Array [],
      "nodDate": "2020-08-05",
      "overtime": false,
      "readableHearingRequestType": "Video",
      "readableOriginalHearingRequestType": "Video",
      "regionalOffice": null,
      "removed": false,
      "status": "not_distributed",
      "vacateType": null,
      "veteranDateOfDeath": null,
      "veteranFileNumber": "123456789",
      "veteranFullName": "Abellona Valtas",
      "veteranInfo": Object {
        "veteran": Object {
          "address": Object {
            "address_line_1": "1234 Main Street",
            "address_line_2": null,
            "address_line_3": null,
            "city": "Orlando",
            "country": "USA",
            "state": "FL",
            "zip": "12345",
          },
          "date_of_birth": "01/10/1935",
          "date_of_death": null,
          "email_address": "Abellona.Valtas@test.com",
          "full_name": "Abellona Valtas",
          "gender": "M",
        },
      },
      "withdrawn": false,
    }
  }
  hearingType="Central"
>
  <Connect(HearingTypeConversionModal)
    appeal={
      Object {
        "appellantAddress": Object {
          "address_line_1": "9999 MISSION ST",
          "address_line_2": "UBER",
          "address_line_3": "APT 2",
          "city": "SAN FRANCISCO",
          "country": "USA",
          "state": "CA",
          "zip": "94103",
        },
        "appellantFullName": "Abellona Valtas",
        "appellantIsNotVeteran": false,
        "appellantRelationship": "Spouse",
        "appellantTz": "America/New_York",
        "assignedAttorney": null,
        "assignedJudge": null,
        "assignedToLocation": "Hearing Admin",
        "attorneyCaseRewriteDetails": Object {
          "note_from_attorney": null,
          "untimely_evidence": null,
        },
        "availableHearingLocations": Array [],
        "canEditDocumentId": false,
        "canEditRequestIssues": false,
        "caseReviewId": null,
        "caseType": "Original",
        "caseflowVeteranId": 541,
        "certificationDate": null,
        "closestRegionalOffice": null,
        "closestRegionalOfficeLabel": null,
        "completedHearingOnPreviousAppeal": false,
        "decisionDate": null,
        "decisionIssues": Array [
          Object {
            "description": "This is a description of the decision",
            "id": 1,
          },
          Object {
            "description": "This is a description of another decision",
            "id": 2,
          },
        ],
        "docketName": "hearing",
        "docketNumber": "200805-541",
        "documentID": null,
        "isAdvancedOnDocket": false,
        "isLegacyAppeal": false,
        "isPaperCase": false,
        "issueCount": 0,
        "issues": Array [],
        "nodDate": "2020-08-05",
        "overtime": false,
        "readableHearingRequestType": "Video",
        "readableOriginalHearingRequestType": "Video",
        "regionalOffice": null,
        "removed": false,
        "status": "not_distributed",
        "vacateType": null,
        "veteranDateOfDeath": null,
        "veteranFileNumber": "123456789",
        "veteranFullName": "Abellona Valtas",
        "veteranInfo": Object {
          "veteran": Object {
            "address": Object {
              "address_line_1": "1234 Main Street",
              "address_line_2": null,
              "address_line_3": null,
              "city": "Orlando",
              "country": "USA",
              "state": "FL",
              "zip": "12345",
            },
            "date_of_birth": "01/10/1935",
            "date_of_death": null,
            "email_address": "Abellona.Valtas@test.com",
            "full_name": "Abellona Valtas",
            "gender": "M",
          },
        },
        "withdrawn": false,
      }
    }
    hearingType="Central"
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    location={
      Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      }
    }
    match={
      Object {
        "isExact": true,
        "params": Object {},
        "path": "/",
        "url": "/",
      }
    }
  >
    <HearingTypeConversionModal
      clearAppealDetails={[Function]}
      hearingType="Central"
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
      location={
        Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        }
      }
      match={
        Object {
          "isExact": true,
          "params": Object {},
          "path": "/",
          "url": "/",
        }
      }
      showErrorMessage={[Function]}
      showSuccessMessage={[Function]}
    >
      <Modal
        buttons={Array []}
        cancelButton={
          <Button
            classNames={
              Array [
                "cf-submit",
              ]
            }
            disabled={false}
            linkStyling={true}
            loading={false}
            onClick={[Function]}
            type="button"
            willNeverBeLoading={false}
          >
            Cancel
          </Button>
        }
        className=""
        closeHandler={[Function]}
        confirmButton={
          <Button
            classNames={
              Array [
                "cf-submit",
              ]
            }
            disabled={false}
            linkStyling={false}
            loading={false}
            onClick={[Function]}
            type="button"
            willNeverBeLoading={false}
          >
            Convert Hearing to 
            Central
          </Button>
        }
        scrollLock={true}
        title="Convert Hearing To Central"
      >
        <section
          aria-describedby="modal_id-desc"
          aria-labelledby="modal_id-title"
          aria-modal="true"
          className="cf-modal active "
          id="modal_id"
          role="dialog"
        >
          <TouchProvider>
            <ScrollLock
              accountForScrollbars={true}
            >
              <Sheet
                styles="body {
      box-sizing: border-box !important;
      overflow: hidden !important;
      position: relative !important;
      
      padding-right: 1024px !important;
    }"
              />
            </ScrollLock>
          </TouchProvider>
          <div
            className="cf-modal-body"
            id=""
          >
            <button
              className="cf-modal-close"
              id="Convert-Hearing-To-Central-button-id-close"
              onClick={[Function]}
              type="button"
            >
              <span
                className="usa-sr-only"
              >
                Close
              </span>
              <CloseIcon
                className="cf-icon-close"
                color="#212121"
                size={55}
              >
                <svg
                  className="cf-icon-close"
                  height={55}
                  viewBox="0 0 55 55"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title>
                    close
                  </title>
                  <g
                    fill="#212121"
                  >
                    <path
                      d="M52.6 46.9l-6 6c-.8.8-1.9 1.2-3 1.2s-2.2-.4-3-1.2l-13-13-13 13c-.8.8-1.9 1.2-3 1.2s-2.2-.4-3-1.2l-6-6c-.8-.8-1.2-1.9-1.2-3s.4-2.2 1.2-3l13-13-13-13c-.8-.8-1.2-1.9-1.2-3s.4-2.2 1.2-3l6-6c.8-.8 1.9-1.2 3-1.2s2.2.4 3 1.2l13 13 13-13c.8-.8 1.9-1.2 3-1.2s2.2.4 3 1.2l6 6c.8.8 1.2 1.9 1.2 3s-.4 2.2-1.2 3l-13 13 13 13c.8.8 1.2 1.9 1.2 3s-.4 2.2-1.2 3z"
                    />
                  </g>
                </svg>
              </CloseIcon>
            </button>
            <div
              style={
                Object {
                  "display": "flex",
                }
              }
            >
              <div
                data-css-1gs0ko2=""
              >
                <h1
                  id="modal_id-title"
                >
                  Convert Hearing To Central
                </h1>
                <div
                  data-css-o98ubo=""
                >
                  <p
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "The hearing request will appear in the scheduling queue for the <strong>Central Office</strong>.",
                      }
                    }
                  />
                </div>
              </div>
            </div>
            <div
              className="cf-modal-divider"
            />
            <div
              className="cf-modal-controls"
            >
              <div>
                <span
                  className="cf-push-right"
                >
                  <Button
                    classNames={
                      Array [
                        "cf-submit",
                      ]
                    }
                    disabled={false}
                    linkStyling={false}
                    loading={false}
                    onClick={[Function]}
                    type="button"
                    willNeverBeLoading={false}
                  >
                    <span>
                      <button
                        className="cf-submit usa-button"
                        disabled={false}
                        onClick={[Function]}
                        type="button"
                      >
                        Convert Hearing to 
                        Central
                      </button>
                    </span>
                  </Button>
                </span>
                <span
                  className="cf-push-left"
                >
                  <Button
                    classNames={
                      Array [
                        "cf-submit",
                      ]
                    }
                    disabled={false}
                    linkStyling={true}
                    loading={false}
                    onClick={[Function]}
                    type="button"
                    willNeverBeLoading={false}
                  >
                    <span>
                      <button
                        className="cf-submit cf-btn-link usa-button"
                        disabled={false}
                        onClick={[Function]}
                        type="button"
                      >
                        Cancel
                      </button>
                    </span>
                  </Button>
                </span>
              </div>
            </div>
          </div>
        </section>
      </Modal>
    </HearingTypeConversionModal>
  </Connect(HearingTypeConversionModal)>
</withRouter(Connect(HearingTypeConversionModal))>
`;

exports[`HearingTypeConversion Displays convert to Video text when converting from Video 1`] = `
<withRouter(Connect(HearingTypeConversionModal))
  appeal={
    Object {
      "appellantAddress": Object {
        "address_line_1": "9999 MISSION ST",
        "address_line_2": "UBER",
        "address_line_3": "APT 2",
        "city": "SAN FRANCISCO",
        "country": "USA",
        "state": "CA",
        "zip": "94103",
      },
      "appellantFullName": "Abellona Valtas",
      "appellantIsNotVeteran": false,
      "appellantRelationship": "Spouse",
      "appellantTz": "America/New_York",
      "assignedAttorney": null,
      "assignedJudge": null,
      "assignedToLocation": "Hearing Admin",
      "attorneyCaseRewriteDetails": Object {
        "note_from_attorney": null,
        "untimely_evidence": null,
      },
      "availableHearingLocations": Array [],
      "canEditDocumentId": false,
      "canEditRequestIssues": false,
      "caseReviewId": null,
      "caseType": "Original",
      "caseflowVeteranId": 541,
      "certificationDate": null,
      "closestRegionalOffice": null,
      "closestRegionalOfficeLabel": null,
      "completedHearingOnPreviousAppeal": false,
      "decisionDate": null,
      "decisionIssues": Array [
        Object {
          "description": "This is a description of the decision",
          "id": 1,
        },
        Object {
          "description": "This is a description of another decision",
          "id": 2,
        },
      ],
      "docketName": "hearing",
      "docketNumber": "200805-541",
      "documentID": null,
      "isAdvancedOnDocket": false,
      "isLegacyAppeal": false,
      "isPaperCase": false,
      "issueCount": 0,
      "issues": Array [],
      "nodDate": "2020-08-05",
      "overtime": false,
      "readableHearingRequestType": "Video",
      "readableOriginalHearingRequestType": "Video",
      "regionalOffice": null,
      "removed": false,
      "status": "not_distributed",
      "vacateType": null,
      "veteranDateOfDeath": null,
      "veteranFileNumber": "123456789",
      "veteranFullName": "Abellona Valtas",
      "veteranInfo": Object {
        "veteran": Object {
          "address": Object {
            "address_line_1": "1234 Main Street",
            "address_line_2": null,
            "address_line_3": null,
            "city": "Orlando",
            "country": "USA",
            "state": "FL",
            "zip": "12345",
          },
          "date_of_birth": "01/10/1935",
          "date_of_death": null,
          "email_address": "Abellona.Valtas@test.com",
          "full_name": "Abellona Valtas",
          "gender": "M",
        },
      },
      "withdrawn": false,
    }
  }
  hearingType="Video"
>
  <Connect(HearingTypeConversionModal)
    appeal={
      Object {
        "appellantAddress": Object {
          "address_line_1": "9999 MISSION ST",
          "address_line_2": "UBER",
          "address_line_3": "APT 2",
          "city": "SAN FRANCISCO",
          "country": "USA",
          "state": "CA",
          "zip": "94103",
        },
        "appellantFullName": "Abellona Valtas",
        "appellantIsNotVeteran": false,
        "appellantRelationship": "Spouse",
        "appellantTz": "America/New_York",
        "assignedAttorney": null,
        "assignedJudge": null,
        "assignedToLocation": "Hearing Admin",
        "attorneyCaseRewriteDetails": Object {
          "note_from_attorney": null,
          "untimely_evidence": null,
        },
        "availableHearingLocations": Array [],
        "canEditDocumentId": false,
        "canEditRequestIssues": false,
        "caseReviewId": null,
        "caseType": "Original",
        "caseflowVeteranId": 541,
        "certificationDate": null,
        "closestRegionalOffice": null,
        "closestRegionalOfficeLabel": null,
        "completedHearingOnPreviousAppeal": false,
        "decisionDate": null,
        "decisionIssues": Array [
          Object {
            "description": "This is a description of the decision",
            "id": 1,
          },
          Object {
            "description": "This is a description of another decision",
            "id": 2,
          },
        ],
        "docketName": "hearing",
        "docketNumber": "200805-541",
        "documentID": null,
        "isAdvancedOnDocket": false,
        "isLegacyAppeal": false,
        "isPaperCase": false,
        "issueCount": 0,
        "issues": Array [],
        "nodDate": "2020-08-05",
        "overtime": false,
        "readableHearingRequestType": "Video",
        "readableOriginalHearingRequestType": "Video",
        "regionalOffice": null,
        "removed": false,
        "status": "not_distributed",
        "vacateType": null,
        "veteranDateOfDeath": null,
        "veteranFileNumber": "123456789",
        "veteranFullName": "Abellona Valtas",
        "veteranInfo": Object {
          "veteran": Object {
            "address": Object {
              "address_line_1": "1234 Main Street",
              "address_line_2": null,
              "address_line_3": null,
              "city": "Orlando",
              "country": "USA",
              "state": "FL",
              "zip": "12345",
            },
            "date_of_birth": "01/10/1935",
            "date_of_death": null,
            "email_address": "Abellona.Valtas@test.com",
            "full_name": "Abellona Valtas",
            "gender": "M",
          },
        },
        "withdrawn": false,
      }
    }
    hearingType="Video"
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    location={
      Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      }
    }
    match={
      Object {
        "isExact": true,
        "params": Object {},
        "path": "/",
        "url": "/",
      }
    }
  >
    <HearingTypeConversionModal
      clearAppealDetails={[Function]}
      hearingType="Video"
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
      location={
        Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        }
      }
      match={
        Object {
          "isExact": true,
          "params": Object {},
          "path": "/",
          "url": "/",
        }
      }
      showErrorMessage={[Function]}
      showSuccessMessage={[Function]}
    >
      <Modal
        buttons={Array []}
        cancelButton={
          <Button
            classNames={
              Array [
                "cf-submit",
              ]
            }
            disabled={false}
            linkStyling={true}
            loading={false}
            onClick={[Function]}
            type="button"
            willNeverBeLoading={false}
          >
            Cancel
          </Button>
        }
        className=""
        closeHandler={[Function]}
        confirmButton={
          <Button
            classNames={
              Array [
                "cf-submit",
              ]
            }
            disabled={false}
            linkStyling={false}
            loading={false}
            onClick={[Function]}
            type="button"
            willNeverBeLoading={false}
          >
            Convert Hearing to 
            Video
          </Button>
        }
        scrollLock={true}
        title="Convert Hearing To Video"
      >
        <section
          aria-describedby="modal_id-desc"
          aria-labelledby="modal_id-title"
          aria-modal="true"
          className="cf-modal active "
          id="modal_id"
          role="dialog"
        >
          <TouchProvider>
            <ScrollLock
              accountForScrollbars={true}
            >
              <Sheet
                styles="body {
      box-sizing: border-box !important;
      overflow: hidden !important;
      position: relative !important;
      
      padding-right: 1024px !important;
    }"
              />
            </ScrollLock>
          </TouchProvider>
          <div
            className="cf-modal-body"
            id=""
          >
            <button
              className="cf-modal-close"
              id="Convert-Hearing-To-Video-button-id-close"
              onClick={[Function]}
              type="button"
            >
              <span
                className="usa-sr-only"
              >
                Close
              </span>
              <CloseIcon
                className="cf-icon-close"
                color="#212121"
                size={55}
              >
                <svg
                  className="cf-icon-close"
                  height={55}
                  viewBox="0 0 55 55"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title>
                    close
                  </title>
                  <g
                    fill="#212121"
                  >
                    <path
                      d="M52.6 46.9l-6 6c-.8.8-1.9 1.2-3 1.2s-2.2-.4-3-1.2l-13-13-13 13c-.8.8-1.9 1.2-3 1.2s-2.2-.4-3-1.2l-6-6c-.8-.8-1.2-1.9-1.2-3s.4-2.2 1.2-3l13-13-13-13c-.8-.8-1.2-1.9-1.2-3s.4-2.2 1.2-3l6-6c.8-.8 1.9-1.2 3-1.2s2.2.4 3 1.2l13 13 13-13c.8-.8 1.9-1.2 3-1.2s2.2.4 3 1.2l6 6c.8.8 1.2 1.9 1.2 3s-.4 2.2-1.2 3l-13 13 13 13c.8.8 1.2 1.9 1.2 3s-.4 2.2-1.2 3z"
                    />
                  </g>
                </svg>
              </CloseIcon>
            </button>
            <div
              style={
                Object {
                  "display": "flex",
                }
              }
            >
              <div
                data-css-1gs0ko2=""
              >
                <h1
                  id="modal_id-title"
                >
                  Convert Hearing To Video
                </h1>
                <div
                  data-css-o98ubo=""
                >
                  <p
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "The hearing request will appear in the scheduling queue for the appropriate regional office.",
                      }
                    }
                  />
                </div>
              </div>
            </div>
            <div
              className="cf-modal-divider"
            />
            <div
              className="cf-modal-controls"
            >
              <div>
                <span
                  className="cf-push-right"
                >
                  <Button
                    classNames={
                      Array [
                        "cf-submit",
                      ]
                    }
                    disabled={false}
                    linkStyling={false}
                    loading={false}
                    onClick={[Function]}
                    type="button"
                    willNeverBeLoading={false}
                  >
                    <span>
                      <button
                        className="cf-submit usa-button"
                        disabled={false}
                        onClick={[Function]}
                        type="button"
                      >
                        Convert Hearing to 
                        Video
                      </button>
                    </span>
                  </Button>
                </span>
                <span
                  className="cf-push-left"
                >
                  <Button
                    classNames={
                      Array [
                        "cf-submit",
                      ]
                    }
                    disabled={false}
                    linkStyling={true}
                    loading={false}
                    onClick={[Function]}
                    type="button"
                    willNeverBeLoading={false}
                  >
                    <span>
                      <button
                        className="cf-submit cf-btn-link usa-button"
                        disabled={false}
                        onClick={[Function]}
                        type="button"
                      >
                        Cancel
                      </button>
                    </span>
                  </Button>
                </span>
              </div>
            </div>
          </div>
        </section>
      </Modal>
    </HearingTypeConversionModal>
  </Connect(HearingTypeConversionModal)>
</withRouter(Connect(HearingTypeConversionModal))>
`;
